mslib/msui/multiple_flightpath_dockwidget.py

```diff
@@ -84,185 +105,423 @@ class MultipleFlightpathControlWidget(QtWidgets.QWidget, ui.Ui_MultipleViewWidge
+                # if self.dict_flighttrack[listItem.flighttrack_model]["checkState"]:
+                #     listItem.setCheckState(QtCore.Qt.Checked)
+            self.set_activate_flag()
+            listItem.setFont(font)
+
+    def drawInactiveFlighttracks(self, list_widget):
         """
-        Add flighttracks to list widget
+        Draw inactive flighttracks
         """
-        wp_model = ft.WaypointsTableModel(filename=filename)
-        for count in range(self.listView.count()):
-            if str(self.listView.item(count).flighttrack_model) == str(wp_model):
-                break
+        for entry in self.dict_flighttrack.values():
+            if entry["patch"] is not None:
+                entry["patch"].remove()
+
+        for index in range(list_widget.count()):
+            listItem = list_widget.item(index)
+            if hasattr(list_widget.item(index), "checkState") and (
+                    list_widget.item(index).checkState() == QtCore.Qt.Checked):
+                if listItem.flighttrack_model != self.active_flight_track:
+                    patch = MultipleFlightpath(self.view.map,
+                                               self.dict_flighttrack[listItem.flighttrack_model][
+                                                   "wp_data"],
+                                               color=self.dict_flighttrack[listItem.flighttrack_model]['color'])
+
+                    self.dict_flighttrack[listItem.flighttrack_model]["patch"] = patch
+                    self.dict_flighttrack[listItem.flighttrack_model]["checkState"] = True
+            else:
+                # pass
+                self.dict_flighttrack[listItem.flighttrack_model]["checkState"] = False
+
+    def set_activate_flag(self):
+        if not self.flighttrack_added:
+            self.flighttrack_activated = True
+
+    @QtCore.Slot(int)
+    def update_op_id(self, op_id):
+        self.operations.get_op_id(op_id)
+
+
+class MultipleFlightpathOperations:
+    """
+    This class provides the functions for plotting Multiple Flighttracks from mscolab server
+    on the TopView canvas.
+    """
+
+    def __init__(self, mscolab_server_url, token, list_operation_track, listOperationsMSC, view):
+        # Variables related to Mscolab Operations
+        self.active_op_id = None
+        self.mscolab_server_url = mscolab_server_url
+        self.token = token
+        self.dict_operations = {}
+        self.list_operation_track = list_operation_track
+        self.listOperationsMSC = listOperationsMSC
+        self.active_flight_track = None
+        self.view = view
+
+        self.operation_added = False
+        self.operation_removed = False
+        self.operation_activated = False
+
+        # Connect signals and slots
+        self.list_operation_track.itemChanged.connect(self.set_flag)
+        self.listOperationsMSC.model().rowsInserted.connect(self.wait2)
+        self.listOperationsMSC.model().rowsRemoved.connect(self.operationRemoved)
```

Comment:
On removing operations, `rowsInserted()` signal is also emitted and on inserting operations -- `rowsRemoved()` signal is also emitted